server/src/main/java/org/opensearch/rest/action/list/RestIndicesListAction.java

```
@@ -0,0 +1,104 @@
+        return unmodifiableList(asList(new Route(GET, "/_list/indices"), new Route(GET, "/_list/indices/{index}")));
+    }
+
+    @Override
+    public String getName() {
+        return "list_indices_action";
+    }
+
+    @Override
+    protected void documentation(StringBuilder sb) {
+        sb.append("/_list/indices\n");
+        sb.append("/_list/indices/{index}\n");
+    }
+
+    @Override
+    public boolean isActionPaginated() {
+        return true;
+    }
+
+    @Override
+    protected PageParams validateAndGetPageParams(RestRequest restRequest) {
+        PageParams pageParams = super.validateAndGetPageParams(restRequest);
+        // validate max supported pageSize
+        if (pageParams.getSize() > MAX_SUPPORTED_LIST_INDICES_PAGE_SIZE_STRING) {
+            throw new IllegalArgumentException("size should be less than [" + MAX_SUPPORTED_LIST_INDICES_PAGE_SIZE_STRING + "]");
+        }
+        // Next Token in the request will be validated by the IndexStrategyTokenParser itself.
+        if (Objects.nonNull(pageParams.getRequestedToken())) {
+            IndexPaginationStrategy.IndexStrategyToken.validateIndexStrategyToken(pageParams.getRequestedToken());
+        }
+        return pageParams;
+    }
+
+    protected int defaultPageSize() {
+        return DEFAULT_LIST_INDICES_PAGE_SIZE_STRING;
+    }
+
+    @Override
+    protected IndexPaginationStrategy getPaginationStrategy(ClusterStateResponse clusterStateResponse) {
+        return new IndexPaginationStrategy(pageParams, clusterStateResponse.getState());
+    }
+
+    // Public for testing
+    @Override
+    public Iterator<Tuple<String, Settings>> getTableIterator(String[] indices, Map<String, Settings> indexSettingsMap) {
+        return new Iterator<>() {
+            int indexPos = 0;
+
+            @Override
+            public boolean hasNext() {
+                while (indexPos < indices.length && indexSettingsMap.containsKey(indices[indexPos]) == false) {
```

we silently skip an index if it is not present in map?

---

ok